File: backend-mcp/src/test/resources/mini-repos/javascript/src/repository.js
export class UserRepository {
  constructor() {
    this._last = null;
  }

  /** Saves a user name and returns the uppercase variant. */
  save(name) {
    this._last = name;
    return typeof name === "string" ? name.toUpperCase() : "";
  }

  last() {
    return this._last;
  }
}

File: backend-mcp/src/test/resources/mini-repos/go/cmd/demo/repository.go
package main

import "fmt"

type Repository interface {
	Save(name string)
}

type MemoryRepository struct {
	last string
}

func (m *MemoryRepository) Save(name string) {
	m.last = name
	fmt.Println("saved", name)
}

func (m *MemoryRepository) Last() string {
	return m.last
}
